front50-migrations/src/main/java/com/netflix/spinnaker/front50/migrations/MoveSubpathToLocationMigration.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2020 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.front50.migrations;
+
+import com.google.common.base.Strings;
+import com.netflix.spinnaker.front50.model.ItemDAO;
+import com.netflix.spinnaker.front50.model.pipeline.Pipeline;
+import com.netflix.spinnaker.front50.model.pipeline.PipelineDAO;
+import java.time.Clock;
+import java.util.*;
+import java.util.function.Predicate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class MoveSubpathToLocationMigration implements Migration {
+
+  private static final Logger log = LoggerFactory.getLogger(CloudProvidersStringMigration.class);
+
+  // Only valid until October 1st, 2020
+  private static final Date VALID_UNTIL = new GregorianCalendar(2020, 10, 1).getTime();
```

Comment:
Also, I'd strongly recommend passing in `Calendar.OCTOBER` for the month (or whichever month you choose)...as this is actually November 1st, 2020 because the month argument is indexed from 0.